DSEGOG-350 Delete favourite filter

joshdimanteto opened the pull request:
## Description
delete a favourite filter 

## Testing instructions


- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
closes DSEGOG-350

joshdimanteto committed:
DSEGOG-350 Delete favourite filter

joshdimanteto committed:
Merge branch 'DSEGOG-349-edit-favourite-filter' into DSEGOG-350-delete-favourite-filter

louise-davies commented on src/api/favouriteFilters.test.tsx:
```diff
@@ -65,4 +66,25 @@ describe('favourite filters api functions', () => {
+      expect(result.current.data).toEqual('');
+    });
+
+    it.todo(
+      'sends axios request to delete user favourite Filter and throws an appropriate error on failure'
+    );
```
Since this is a mutation that actually has error handling attached to it already, the test doesn't need to be a TODO

louise-davies commented on src/mocks/handlers.ts:
```diff
@@ -381,6 +381,15 @@ export const handlers = [
+  http.delete('/users/filters/:id', async ({ params }) => {
+    const { id } = params;
+
+    if (id === '66f2e06a41a3fe93d6c81395') {
```
Probably useful to include a comment on which filter this is referencing (I think this is test 3?)

joshdimanteto committed:
Address review comments

- remove todos for mutations react query hooks
- use the favouriteFiltersJson to get the id

louise-davies reviewed (approved):
LGTM
